### PR TITLE
Fix completion in the presence of multiple lines

### LINKED ISSUE #1528: Deleting the following lines when doing TAB completition above
Steps to reproduce:

In one cell:
```
def a(arg):
    pass
```
in the second:
```
a(ar)
b()
```
Go to first line of the second:
image
press TAB to autocomplete argument `arg` and it deletes all the following lines (`b()` in this case):
image

## EDIT
OK, not just following lines, but all lines in the cell except the one you do autocomplete on. E.g. pressing TAB in this situation:
image

gives me:
image

### PR DESCRIPTION
**Pull request by blink1073**
Regression introduced in #1467.  Fixes #1528.

### CODE REVIEW

File: src/completer/handler.ts [hunk 1]
```diff
@@ -202,16 +202,26 @@ class CompletionHandler implements IDisposable {
    * Handle a completion selected signal from the completion widget.
    */
   protected onCompletionSelected(widget: CompleterWidget, value: string): void {
-    if (!this._editor || !this._completer.model) {
+    let model = this._completer.model;
+    let editor = this._editor;
+    if (!this._editor || !model) {
```
Comment (jasongrout) [inline src/completer/handler.ts]: `!editor`

File: test/src/completer/handler.spec.ts [hunk 2]
```diff
@@ -444,11 +444,11 @@ describe('completer/handler', () => {
         };
 
         handler.editor = editor;
-        handler.editor.model.value.text = 'foo';
+        handler.editor.model.value.text = 'foo\nbaz';
```
Comment (jasongrout) [inline test/src/completer/handler.spec.ts]: Can we add a line above foo as well?

**Comment by blink1073** [comment]
@MeeseeksDev backport to 0.13.x

**Comment by blink1073** [comment]
@meeseeksdev backport to 0.13.x

**Comment by blink1073** [comment]
@meeseeksdev hello

**Comment by blink1073** [comment]
@meeseeksdev hello

**Comment by blink1073** [comment]
@meeseeksdev backport to 0.13.x

**Comment by blink1073** [comment]
@meeseeksdev hello

**Comment by blink1073** [comment]
@meeseeksdev backport to 0.13.x

**Comment by blink1073** [comment]
@Carreau, am I missing something that would make the backport feature work?  I tried open and close successfully in another issue.

**Comment by Carreau** [comment]
Hum, it might have failed the backport let me check the logs.

**Comment by Carreau** [comment]
Too far in the logs, let me retry.

@meeseeksdev backport to 0.13.x

**Comment by Carreau** [comment]
Hum, it say there is nothing to backport (empty commit), and crash. I'll fix the crahs and investigate empty commit.

**Comment by Carreau** [comment]
I'm confused, master and 0.13.x have 2 commits of difference and this was already in 0.13.x I'll still fix the bot to reply with a message.

**Comment by blink1073** [comment]
Argh!  I had tried to make a branch that started at the last git tag, sorry for the noise.

**Comment by willingc** [comment]
@Carreau Perhaps the bot is trying to take over the 🌏

**Comment by Carreau** [comment]
> Argh! I had tried to make a branch that started at the last git tag, sorry for the noise.

Well, you can still force-push a new 0.13.x or master branch if needed :-)

**Comment by blink1073** [comment]
@meeseeksdev backport to 0.13.x

**Comment by blink1073** [comment]
Booyah!

**Comment by jasongrout** [comment]
meeseeksdev - finish JupyterLab!

**Comment by Carreau** [comment]
> meeseeksdev - finish JupyterLab!

Not implemented yet. Also you were not on the whitelist. 
Though you can try the following. 

@meeseeksdev[bot] take over the world

**Comment by Carreau** [comment]
(it uses the PR  or issue author)

**Comment by blink1073** [comment]
Nice.  This is really cool, @Carreau, kudos!

magic

**Comment by jasongrout** [comment]
Am I on the whitelist now?

@meeseeksdev[bot] take over the world

**Comment by Carreau** [comment]
@jasongrout yes you are on the whitelist, and sudo not yet implemented :-) 

Pr to do random quotes of Pinky and Brain (or other) welcome.

**Comment by blink1073** [comment]
"Brain: Pinky, if I had arms, and wasn't a ball of glowing hydrogen, I would hurt you."

**Comment by willingc** [comment]
Oh my goodness @Carreau, the bot watches Pinky and the Brain. Loved that show. How about MST3000?

**Comment by Carreau** [comment]
Unfortunately not all show where translated to French (and the bot is
natively French) so was not raised knowing all of these show, or learned
the French quotes of the main characters.

The bot will be happy to get a PR to increase its knowledge of quotes.

**Pull request merged**